README.md

```diff
@@ -2,31 +2,322 @@
+value is an even number. The `result` local variable is incorrectly initialized
+to `true` rather than `false`.  The method should return `false`
+if the parameter is odd, but the method always returns `true` due to
+the incorrect initialization.
+
+The Junit test class `BuggyExample2Test` calls the method once, passing
+an even number `8` as a parameter.
+
+```java
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class BuggyExample2Test {
+    @Test
+    public void isEven() {
+        assertTrue(BuggyExample2.isEven(8));
+    }
+
+}
+```
+
+Right-click on the Junit class `BuggyExample2Test` and select `Run BuggyExample2Test with Coverage`.
+
+Notice we achieve 100% line coverage of the `BuggyExample2` class and the test
+passes.
+
+![run coverage step 4](https://curriculum-content.s3.amazonaws.com/6676/coverage/run_coverage4.png)
+
+Since we know where is an error in the code, this tells us we need a stronger
+measure of coverage to find our error.
+
+Let's consider the possible execution paths through the `isEven` method.
+The method has 4 executable lines of code numbered 4, 5, 6, and 7:
+
+![iseven method with line numbers](https://curriculum-content.s3.amazonaws.com/6676/coverage/iseven_method.png)
+
+If we view the method as a flowchart, we see there are two possible
+execution paths through the code:
+
+- A path executing lines 4, 5, 6, 7 (x % 2 == 0 is true)
+- A path executing lines 4, 5, 7 (x % 2 == 0 is false)
+
+![buggy flowchart2](https://curriculum-content.s3.amazonaws.com/6676/coverage/buggy_flowchart2.png)
+
+
+**Branch coverage** ensures the tests cause all branches
+of all control structures in a program to be executed at least once.
+Branch coverage requires at least one test that causes the boolean condition
+in the `if` statement to evaluate to `true`, and at least one test that causes it to
+evaluate to `false`.
+
+### Jacoco Coverage Runner
+
+The default IntelliJ coverage runner does not include a branch coverage metric.
+However, there is another runner named **Jacoco** (Java Code Coverage) that does.
+
+Right-click on the `BuggyExample2Test` class and select "Modify Run Configuration".
+
+![modify run configuration](https://curriculum-content.s3.amazonaws.com/6676/coverage/modify_run.png)
+
+Click on the link titled "Modify Options".
+
+![modify options](https://curriculum-content.s3.amazonaws.com/6676/coverage/modify_options.png)
+
+Select "Specify alternative coverage runner".
+
+![alternative runner](https://curriculum-content.s3.amazonaws.com/6676/coverage/alternative_runner.png)
+
+Select "Jacoco" and then press the "OK" button.
+
+![jacoco runner](https://curriculum-content.s3.amazonaws.com/6676/coverage/jacoco_runner.png)
+
+Right-click on the Junit class `BuggyExample2Test` and select `Run BuggyExample2Test with Coverage`.
+The coverage view now includes an additional metric for branch coverage.
+We can see only the test only coverage 50% of branches since the
```

Comment:
Reword suggestion:

"We can see the test only covers 50% of the branches since the test never executes the branch where `x % 2 == 0` is `false`."

Reply:
thanks, obviously working too late when that sentence was written :-)